source/module.h

```diff
@@ -69,6 +69,21 @@ napi_status aws_napi_create_threadsafe_function(
     void *context,
     napi_threadsafe_function *result);
 
+/**
+ * Wrapper around napi_release_threadsafe_function,
+ * check the function before releasing it.
+ */
+napi_status aws_napi_release_threadsafe_function(
+    napi_threadsafe_function function,
+    napi_threadsafe_function_release_mode mode);
+
+/**
+ * Wrapper around napi_unref_threadsafe_function,
+ * Incase release the threadsafe function from that function is needed, unref will let env go
+ * and the function will be cleaned up as env clean itself up
```

Comment:
Did an experiment that after unref, the `napi_finalize thread_finalize_cb` will be invoked even if no release ever invoked.🙆‍♂️

Reply:
Note: If you want to release a thread safe function from within that thread safe function's callback, call unref instead, and the function will be finalized later by the environment.